app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -123,34 +120,26 @@ class SearchRepositoriesActivity : AppCompatActivity() {
         binding.searchRepo.text.trim().let {
             if (it.isNotEmpty()) {
                 binding.list.scrollToPosition(0)
-                viewModel.searchRepo(it.toString())
-                adapter.submitList(null)
+                search(it.toString())
             }
         }
     }
 
-    private fun showEmptyList(show: Boolean) {
-        if (show) {
-            binding.emptyList.visibility = View.VISIBLE
-            binding.list.visibility = View.GONE
-        } else {
-            binding.emptyList.visibility = View.GONE
-            binding.list.visibility = View.VISIBLE
+    private fun search(query: String) {
```

Comment:
Would be nice if we could represent this as a `Flow<String>` and then we could just do collectLatest / mapLatest on it !!